src/hooks/useTaskManager.ts

```diff
@@ -0,0 +1,81 @@
+import { useEffect, useState } from 'react';
+import { Task } from '../utils/types';
+import {
+  addTask,
+  deleteAllTasks,
+  deleteSingleTask,
+  getTasks,
+  updateTask,
+} from '../utils/api';
+
+function useTaskManager() {
+  const [tasks, setTasks] = useState<Task[]>([]);
+
+  useEffect(() => {
+    fetchTasks()
+      .then(() => {})
+      .catch((err) => {
+        console.error('Error getting tasks: ', err);
+      });
+  }, []);
+
+  //Fetch tasks from API
+  const fetchTasks = async () => {
```

Comment:
Question 🤔: why don't we move this outside of this hook and pass it a callback?Will make this easier to test in the future as well.

Reply:
updated to be outside hook and passed as a callback